cpp/src/io/text/multibyte_split.cu

```diff
@@ -0,0 +1,401 @@
+    };
+  } temp_storage;
+
+  int32_t const tile_idx            = base_tile_idx + blockIdx.x;
+  int32_t const tile_input_offset   = blockIdx.x * ITEMS_PER_TILE;
+  int32_t const thread_input_offset = tile_input_offset + threadIdx.x * ITEMS_PER_THREAD;
+  int32_t const thread_input_size   = chunk_input_chars.size() - thread_input_offset;
+
+  // STEP 1: Load inputs
+
+  char thread_chars[ITEMS_PER_THREAD];
+
+  InputLoad(temp_storage.input_load)
+    .Load(chunk_input_chars.data() + tile_input_offset,
+          thread_chars,
+          chunk_input_chars.size() - tile_input_offset);
+
+  // STEP 2: Scan inputs to determine absolute thread states
+
+  uint32_t thread_states[ITEMS_PER_THREAD];
+
+  __syncthreads();                        // required before temp_memory re-use
+  PatternScan(temp_storage.pattern_scan)  //
+    .Scan(tile_idx, tile_multistates, trie, thread_chars, thread_states);
+
+  // STEP 3: Flag matches
+
+  uint32_t thread_offsets[ITEMS_PER_THREAD];
+
+  for (int32_t i = 0; i < ITEMS_PER_THREAD; i++) {
+    thread_offsets[i] = i < thread_input_size and trie.is_match(thread_states[i]);
+  }
+
+  // STEP 4: Scan flags to determine absolute thread output offset
+
+  __syncthreads();  // required before temp_memory re-use
+  auto prefix_callback =
+    OffsetScanCallback(temp_storage.offset_scan_callback, tile_output_offsets, tile_idx);
+
+  OffsetScan(temp_storage.offset_scan)
+    .ExclusiveSum(thread_offsets, thread_offsets, prefix_callback);
+
+  // Step 5: Assign outputs from each thread using match offsets.
+
+  if (abs_output_chars.size() > 0) {
+    for (int32_t i = 0; i < ITEMS_PER_THREAD and i < thread_input_size; i++) {
+      abs_output_chars[chunk_input_offset + thread_input_offset + i] = thread_chars[i];
+    }
+  }
+
+  if (abs_output_delimiter_offsets.size() > 0) {
+    for (int32_t i = 0; i < ITEMS_PER_THREAD and i < thread_input_size; i++) {
+      if (trie.get_match_length(thread_states[i]) > 0) {
+        auto const match_end = base_tile_idx * ITEMS_PER_TILE + thread_input_offset + i + 1;
+        abs_output_delimiter_offsets[thread_offsets[i]] = match_end;
+      }
+    }
+  }
+}
+
+}  // namespace
+
+namespace cudf {
+namespace io {
+namespace text {
+namespace detail {
+
+template <typename T>
+std::unique_ptr<column> create_column(rmm::device_uvector<T>&& values)
+{
+  auto size  = values.size();
+  auto dtype = cudf::data_type{cudf::type_to_id<T>()};
+
+  CUDF_EXPECTS(dtype.id() != type_id::EMPTY, "column type_id cannot be EMPTY");
+
+  return std::make_unique<cudf::column>(dtype, size, values.release(), rmm::device_buffer(), 0);
+}
+
+std::unique_ptr<column> create_char_column(rmm::device_uvector<char>&& values)
+{
+  auto size  = values.size();
+  auto dtype = cudf::data_type{type_id::INT8};
+
+  return std::make_unique<cudf::column>(dtype, size, values.release(), rmm::device_buffer(), 0);
+}
+
+std::unique_ptr<column> create_strings_column(rmm::device_uvector<char>&& chars,
```

Comment:
similarly, this seems to be a move version of an existing factory function. Do we want this in column_factories?

Reply:
@davidwendt I'm creating a strings column from two `device_uvectors`. Any reason this shouldn't be in column_factories?

```c++
std::unique_ptr<column> make_strings_column(
  rmm::device_uvector<char>&& chars,
  rmm::device_uvector<int32_t>&& offsets,
  size_type null_count,
  rmm::device_buffer&& null_mask,
  rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
```

Reply:
This looks problematic since it allows the caller to create the `device_uvector` with the wrong `memory_resource`. Is there some reason the `device_uvector` instances here could not initially be created as columns in the first place?

Reply:
Because I want to work with known types, not type-erased columns.

Reply:
if this factory is taking ownership of existing data, does it need to take a MR?

Reply:
Looks like the factory this function calls accepts an MR and a stream, but neither are used, which makes because logically we are just transferring ownership.

Reply:
The strings columns factories don't match this use case, and adding one that does match this use case will require adding a lot of overhead that is unnecessary for this use case (null mask, null count, num strings arguments, possibly stream and mr).

Reply:
After talking to @davidwendt we are going to add a new strings column factory that looks like this:
```
std::unique_ptr<column> make_strings_column(size_type num_strings,
                                            rmm::device_uvector<size_type>&& offsets,
                                            rmm::device_uvector<char>&& chars,
                                            size_type null_count           = 0,
                                            rmm::device_buffer&& null_mask = {});
```